File: LLMT-backend/app/schemas/inference.py
"""Model inference schemas (Part 6 of jiekou.md)."""

from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class InferenceModelOut(BaseModel):
    model_config = ConfigDict(protected_namespaces=())
    model_code: str
    model_name: str
    version: str
    framework: Optional[str] = None
    tag: Optional[str] = None


class PredictRequest(BaseModel):
    input: str = Field(..., min_length=1)
    parameters: dict = {}


class PredictResponse(BaseModel):
    model_code: str
    output: str
    latency_ms: float


class AsyncJobRequest(BaseModel):
    model_code: str
    input: str = Field(..., min_length=1)
    parameters: dict = {}


class AsyncJobOut(BaseModel):
    job_id: str
    model_code: str
    status: str
    input: str
    output: Optional[str] = None
    created_at: str
    finished_at: Optional[str] = None


class UsageOut(BaseModel):
    model_config = ConfigDict(protected_namespaces=())
    model_code: str
    total_calls: int
    remaining_calls: int
    limit_per_minute: int
    reset_at: str

File: LLMT-frontend/src/api/inference.ts
import { get, post, type ApiMessage, unwrap } from '@/api/http'

export interface InferenceModel {
  model_code: string
  model_name: string
  version: string
  framework?: string | null
  tag?: string | null
}

export interface PredictResult {
  model_code: string
  output: string
  latency_ms: number
}

export interface InferenceJob {
  job_id: string
  model_code: string
  status: string
  input: string
  output?: string | null
  created_at: string
  finished_at?: string | null
}

export interface InferenceUsage {
  model_code: string
  total_calls: number
  remaining_calls: number
  limit_per_minute: number
  reset_at: string
}

export const listInferenceModels = async () => unwrap(await get<ApiMessage<InferenceModel[]>>('/inference/models'))

export const predict = async (modelCode: string, payload: { input: string; parameters?: Record<string, unknown> }) =>
  unwrap(await post<ApiMessage<PredictResult>>(`/inference/models/${modelCode}/predict`, payload))

export const createInferenceJob = async (payload: { model_code: string; input: string; parameters?: Record<string, unknown> }) =>
  unwrap(await post<ApiMessage<InferenceJob>>('/inference/jobs', payload))

export const getInferenceJob = async (jobId: string) => unwrap(await get<ApiMessage<InferenceJob>>(`/inference/jobs/${jobId}`))

export const cancelInferenceJob = async (jobId: string) => unwrap(await post<ApiMessage<unknown>>(`/inference/jobs/${jobId}/cancel`))

export const getInferenceUsage = async (modelCode?: string) =>
  unwrap(await get<ApiMessage<InferenceUsage[]>>('/inference/usage', { model_code: modelCode }))
